num/__private/signed_integer_macros.h

```diff
@@ -0,0 +1,445 @@
+    return -primitive_value;                                             \
+  }                                                                      \
+  /** sus::concepts::BitNot trait. */                                    \
+  constexpr inline T operator~()&& noexcept { return ~primitive_value; } \
+  static_assert(true)
+
+#define _sus__signed_binary_logic_ops(T)                                 \
+  /** sus::concepts::Add<##T##> trait. */                                \
+  friend constexpr inline T operator+(const T& l, const T& r) noexcept { \
+    ::sus::check(__private::can_add_without_overflow(l, r));             \
+    return l.primitive_value + r.primitive_value;                        \
+  }                                                                      \
+  /** sus::concepts::Sub<##T##> trait. */                                \
+  friend constexpr inline T operator-(const T& l, const T& r) noexcept { \
+    ::sus::check(__private::can_sub_without_overflow(l, r));             \
+    return l.primitive_value - r.primitive_value;                        \
+  }                                                                      \
+  /** sus::concepts::Mul<##T##> trait. */                                \
+  friend constexpr inline T operator*(const T& l, const T& r) noexcept { \
+    ::sus::check(__private::can_mul_without_overflow(l, r));             \
+    return l.primitive_value * r.primitive_value;                        \
+  }                                                                      \
+  /** sus::concepts::Div<##T##> trait. */                                \
+  friend constexpr inline T operator/(const T& l, const T& r) noexcept { \
+    ::sus::check(__private::can_div_without_overflow(l, r));             \
+    return l.primitive_value / r.primitive_value;                        \
+  }                                                                      \
+  /** sus::concepts::Rem<##T##> trait. */                                \
+  friend constexpr inline T operator%(const T& l, const T& r) noexcept { \
+    ::sus::check(__private::can_div_without_overflow(l, r));             \
+    return l.primitive_value % r.primitive_value;                        \
+  }                                                                      \
+  static_assert(true)
+
+#define _sus__signed_binary_bit_ops(T)                                        \
+  /** sus::concepts::BitAnd<##T##> trait. */                                  \
+  friend constexpr inline T operator&(const T& l, const T& r) noexcept {      \
+    return l.primitive_value & r.primitive_value;                             \
+  }                                                                           \
+  /** sus::concepts::BitOr<##T##> trait. */                                   \
+  friend constexpr inline T operator|(const T& l, const T& r) noexcept {      \
+    return l.primitive_value & r.primitive_value;                             \
```

Comment:
Bug: Should be `|`

Reply:
Thank you! I didn't get to testing this one yet, hopefully I'll cover that hehe :)

Reply:
Fixed